dcpy/utils/s3.py

```diff
@@ -89,33 +119,26 @@ def list_objects(bucket: str, prefix: str) -> list:
     return objects
 
 
-def download_file(
-    bucket: str,
-    key: str,
-    path: Path,
-) -> None:
-    """Downloads a file from S3"""
-    with _progress() as progress:
-        size = get_metadata(bucket, key)["content-length"]
-        task = progress.add_task(
-            f"[green]Downloading [bold]{Path(key).name}[/bold]", total=size
-        )
-        client().download_file(
-            Bucket=bucket,
-            Key=key,
-            Filename=str(path),
-            Callback=lambda bytes: progress.update(task, advance=bytes),
-        )
+def exists(bucket: str, key: str) -> bool:
+    """Returns true if an object with given bucket and key exists"""
+    try:
+        client().head_object(Bucket=bucket, Key=key)
+        return True
+    except ClientError:
```

Comment:
Curious which cases would lead to a ClientError? Do you happen to recall?

Reply:
If the object doesn't exist!

Reply:
Can see if this is a bit too permissive - did some searching and this seemed to be many's recommended solution